src/mistral_common/tokens/tokenizers/sentencepiece.py

```diff
@@ -173,6 +184,25 @@ def to_string(self, tokens: List[int]) -> str:
 
         return text
 
+    def to_string(self, tokens: List[int]) -> str:
+        r"""[DEPRECATED] Converts a list of token ids into a string, keeping special tokens.
+
+        Use `decode` with `special_token_policy=SpecialTokenPolicy.KEEP` instead.
+
+        This is a convenient method for debugging.
+        """
+        warnings.warn(
+            (
+                "`to_string` is deprecated and will be removed in 1.7.0. "
+                "Use `decode` with `special_token_policy=SpecialTokenPolicy.KEEP` instead."
+            ),
+            DeprecationWarning,
```

Comment:
```suggestion
            FutureWarning,
```

is better because it throws a warning by default